### PR TITLE
Set n_validation_batches to 0

### LINKED ISSUE #441: O validation items
**Describe the bug**
```
n. | /home/peter/Github/nowcasting_dataset/nowcasting_dataset/manager.py#L185
DEBUG:nowcasting_dataset.manager:Creating 10 batches x 32 examples per batch = 320 examples for validation.
Traceback (most recent call last):
  File "/home/peter/Github/nowcasting_dataset/scripts/prepare_ml_data.py", line 79, in <module>
    main()
  File "/home/peter/miniconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/peter/miniconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/peter/miniconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/peter/miniconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/peter/Github/nowcasting_dataset/nowcasting_dataset/utils.py", line 155, in inner_func
    return func(*args, **kwargs)
  File "/home/peter/Github/nowcasting_dataset/scripts/prepare_ml_data.py", line 71, in main
    manager.create_files_specifying_spatial_and_temporal_locations_of_each_example_if_necessary()
  File "/home/peter/Github/nowcasting_dataset/nowcasting_dataset/manager.py", line 189, in create_files_specifying_spatial_and_temporal_locations_of_each_example_if_necessary
    df_of_locations = self.sample_spatial_and_temporal_locations_for_examples(
  File "/home/peter/Github/nowcasting_dataset/nowcasting_dataset/manager.py", line 290, in sample_spatial_and_temporal_locations_for_examples
    shuffled_t0_datetimes = np.random.choice(t0_datetimes, size=n_examples)
  File "mtrand.pyx", line 915, in numpy.random.mtrand.RandomState.choice
ValueError: 'a' cannot be empty unless no samples are taken
```

**To Reproduce**
run script preapare_ml_data.py

### PR DESCRIPTION
**Pull request by JackKelly**
# Pull Request

## Description

Related to #426
Fixes #441

Still TODO: Merge in Peter's unit tests for this!

## How Has This Been Tested?

- [ ] No
- [x] Yes, unit tests pass
- [x] `prepare_ml_data.py` runs

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by JackKelly** [comment]
Still TODO: Merge in Peter's unit tests for this!

**Comment by JackKelly** [comment]
Hi @peterdudfield .... so... this is maybe a bit awkward :slightly_smiling_face: 

This PR fixes #441 and maybe slightly addresses #426.  I think this PR fixes the root issue, and also addresses a PR comment you had a while ago about adding descriptions to the `n_<split_name>_batches` fields in `model.py`!

What do you think about merging this PR into `main`?

**Review by peterdudfield — APPROVED** [review]
Looks good, 

I think a test case should be written to make sure this done error - either add it and show an error is made, or add an issue?

File: nowcasting_dataset/manager.py [hunk 1]
```diff
@@ -180,23 +180,38 @@ def create_files_specifying_spatial_and_temporal_locations_of_each_example_if_ne
...
         )
         for split_name, datetimes_for_split in split_t0_datetimes._asdict().items():
-            n_batches = self._get_n_batches_for_split_name(split_name)
-            n_examples = n_batches * self.config.process.batch_size
+            path_for_csv = self.config.output_data.filepath / split_name
+            n_batches_requested = self._get_n_batches_requested_for_split_name(split_name)
+            if (n_batches_requested == 0 and len(datetimes_for_split) != 0) or (
+                len(datetimes_for_split) == 0 and n_batches_requested != 0
+            ):
+                msg = (
```
Comment (peterdudfield) [inline nowcasting_dataset/manager.py:188]: good clear message

I ahve see {variable=} and thanks prints out variable={variable} as a short hand

Comment (JackKelly) [reply nowcasting_dataset/manager.py:188]: ooooh!  `f"{variable=}"` is a really nice trick!  Thanks loads for telling me about that!  I'm gonna use that all over the place now :slightly_smiling_face:

**Comment by JackKelly** [comment]
Good idea about adding a test!  I've added that as issue #450, as I'm keen to get this merged into `main`.  Thanks loads!

**Pull request merged**